Admin Page: Release Announcement Modal

kbrown9 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #13496

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Retrieve the newest Jetpack release post content.
* Display the release post content in the update modal.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is an update to an existing part of Jetpack.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

First, it's important to note that this PR pulls its post data from jetpackreleaseblog.wordpress.com. This currently belongs to goldsounds/@gravityrail, but you can point it at any blog by setting the `JETPACK__RELEASE_POST_BLOG_SLUG` constant. This slug is used to compose the API URL.

Currently that blog ONLY contains a [release post for Jetpack 8.3](https://jetpackreleaseblog.wordpress.com/2020/04/01/test-release-post-8-3/). This post includes text, a link, and an image for testing purposes.

1. Install Jetpack.
2. Apply this branch. 
3. Run `yarn build` to build Jetpack.
4. Temporarily change the Jetpack version to simulate a version update:
   a. In the Jetpack.php file, change the JETPACK__VERSION value to 8.2.
   b. Navigate to wp-admin -> Jetpack.
   c. In the Jetpack.php file, change the JETPACK__VERSION value to 8.3.
   d. Reload the Jetpack dashboard. The update modal containing the release post content for 
       Jetpack 8.3 should display.
5. Change JETPACK__VERSION back to the original value in Jetpack.php.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* TBD

kbrown9 committed:
Display the Jetpack release post content in the upgrade modal.

kbrown9 committed:
Make some improvements to the previous commit:

* Move the preparation of the release post data to a filter on the
 'updating_jetpack_version' hook.
* Add some error handling.

kbrown9 committed:
Code improvements:

* Keep messageContent out of the cookie
* Don't call json_encode() on the release post data. It's no longer necessary since
  the data isn't set in the cookie.
* Don't call JSON.parse() on the localized data. It's not longer necessary since the
  data is sent as an associative array.
* Move the remote request code to a new method and clean it up.

kbrown9 committed:
Some Improvements:
* Add target="_blank" to any links in the post content.
* Change some styling.
* Add links to the release post and the changelog.
* Fix scrolling: only the modal (not the content body) should scroll when
  the modal is displayed.

kbrown9 commented on class.jetpack.php:
```diff
@@ -3188,14 +3189,90 @@ public static function get_activation_source( $referer_url ) {
+			'https://public-api.wordpress.com/rest/v1/sites/jetpack.com/posts'
+		);
+		*/
+
```
We can switch to using the tag in the release post query. We just need to decide on the exact tag that we'll use with the release posts.

jeherve replied:
Would the actual version number be okay, e.g. `7.9`?

kbrown9 replied:
Yes, I think that will be okay!

kbrown9 replied:
This is done - see commit 4d71be6.

kbrown9 commented on class.jetpack.php:
```diff
@@ -3188,14 +3189,90 @@ public static function get_activation_source( $referer_url ) {
+		$post_image_src = isset( $post_data['posts'][0]['featured_image'] ) ? $post_data['posts'][0]['featured_image'] : null;
+		$post_link      = isset( $post_data['posts'][0]['URL'] ) ? $post_data['posts'][0]['URL'] : null;
+
+		$post_content = str_replace( '<a ', '<a target="_blank" rel="noopener noreferrer" ', $post_content );
```
I need to find a better way to do this.

jeherve replied:
What do you think about updating the links later on, in the `UpgradeNoticeContent` component, as you output it on the page? This way it may be easier to avoid adding a duplicate `target` attribute if a link already has one.

kbrown9 replied:
This is done - see commit 4d71be6.

jeherve commented on class.jetpack.php:
```diff
@@ -3188,14 +3189,90 @@ public static function get_activation_source( $referer_url ) {
+		}
+
+		$post_image_src = isset( $post_data['posts'][0]['featured_image'] ) ? $post_data['posts'][0]['featured_image'] : null;
+		$post_link      = isset( $post_data['posts'][0]['URL'] ) ? $post_data['posts'][0]['URL'] : null;
```
Do you think we should bail entirely when we can't find any post link? Or should we take that into account in `renderInnerContent` and not render any `ExternalLink` for that link when it doesn't exist?

kbrown9 replied:
I think we should just bail if we can't find the post link. I made this change in commit 4d71be6.

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -38,62 +42,42 @@ const UpgradeNoticeContent = withModuleSettingsFormHelpers(
-						/>
-					}
-					title={ __( 'New in Jetpack!' ) }
+					svg={ <img src={ this.props.releasePostImage } width="350" alt={ '' } /> }
```
Do we need the alt if we leave it empty?

kbrown9 replied:
I think we do need the empty alt for accessibility. 

This is from the [Alt Text Best Practices](https://make.wordpress.org/accessibility/handbook/content/alternative-text-for-images/#best-practice) in the WordPress Accessibility Handbook:

> Give any decorative images in an img element an empty alt attribute ( alt="" ).

There's some info about why the empty alt tag is important [here](https://www.w3.org/WAI/tutorials/images/decorative/). Basically, the empty alt tag tells the screen reader that the image is decorative. If the alt tag is missing, the screen reader will announce the image's file name.

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -38,62 +42,42 @@ const UpgradeNoticeContent = withModuleSettingsFormHelpers(
+							target="_blank"
+							rel="noopener noreferrer"
+						>
+							{ __( 'View the Jetpack ' + this.props.version + ' changelog' ) }
```
I think something like this would be easier to translate:

```suggestion
							{ __( 'View the Jetpack %(version)s changelog', {
								args: {
									version: this.props.version,
								},
								comment: '%(version) is a version number.',
							} ) }
```

kbrown9 replied:
Thanks! I fixed this in commit 4d71be6.

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -38,62 +42,42 @@ const UpgradeNoticeContent = withModuleSettingsFormHelpers(
-					}
-					title={ __( 'New in Jetpack!' ) }
+					svg={ <img src={ this.props.releasePostImage } width="350" alt={ '' } /> }
+					title={ __( 'New in Jetpack ' + this.props.version + '!' ) }
```
```suggestion
					title={ __( 'New in Jetpack %(version)s', {
						args: {
							version: this.props.version,
						},
						comment: '%(version) is a version number.',
					} ) }
```

kbrown9 replied:
Fixed in commit 4d71be6.

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -38,62 +42,42 @@ const UpgradeNoticeContent = withModuleSettingsFormHelpers(
-							alt={ __( 'Jetpack is ready for the new WordPress editor' ) }
-						/>
-					</p>
+					<div dangerouslySetInnerHTML={ { __html: this.props.releasePostContent } } />
```
I think it may be worth adding a comment here for folks who may stumble across this later. The post content is sanitized before it gets here.

kbrown9 replied:
Good point! I added a comment in commit 4d71be6.

jeherve reviewed (dismissed):
This is pretty awesome, it'll make updates stand out more and make our blog posts more useful from now on. ❤️ 

***********

I wonder if the post image should be in the modal card maybe? I'll let a designer weigh in though, I'm no designer.

![image](https://user-images.githubusercontent.com/426388/67591769-709f8600-f75e-11e9-96cf-53a66e1e5b34.png)


Lists don't appear to be displayed properly right now:

![image](https://user-images.githubusercontent.com/426388/67591545-f2db7a80-f75d-11e9-9ac5-d92c265cfa07.png)

********

Maybe we could make the 3 items in the footer 3 buttons, next to each other. I am curious as to what our designers think about this.

kbrown9 commented:
> I wonder if the post image should be in the modal card maybe? I'll let a designer weigh in though, I'm no designer.

Yeah, that might be better. I'm no designer either 😄 

> Lists don't appear to be displayed properly right now:
> 

Good catch! I added some styling to the li elements, so list elements look like this now:

![Screen Shot on 2019-10-28 at 14-51-01](https://user-images.githubusercontent.com/50059399/67708515-a9c03c00-f992-11e9-811e-cf4c493e492f.png)

(This list is from the [Jetpack 5.9 release post](https://jetpack.com/2018/03/06/jetpack-5-9/)).

> Maybe we could make the 3 items in the footer 3 buttons, next to each other. I am curious as to what our designers think about this.

I think that's also a good idea.

kbrown9 committed:
More improvements:

* Add styling for "li" elements in the post content.
* Add “target” and “rel” attributes to links in the post content in the UpgradeNoticeContent component.
* Fix strings for easier translating.
* Bail if the post_link isn’t available in send_update_modal_data().
* Use version tag to get the correct release post.

kbrown9 commented:
Screenshot of the modal containing the 7.8 release post:

![Screen Shot on 2019-10-28 at 15-00-49](https://user-images.githubusercontent.com/50059399/67710034-8fd42880-f995-11e9-95ec-662b2bfe33d2.png)

MichaelArestad commented:
When we initially designed this feature, it was meant to highlight major changes. The idea was to keep it short and to the point with a link to the release post.

Recommendations:
* Only use it when there is a significant update that highlights a new rad feature or a major change to an existing feature.
* I would use a transparent SVG for the image.
* I wouldn't put any of the credits in there. 

For example, if we announce a significant change to backups or plans, that would be a good time to use this.

dereksmart commented:
@MichaelArestad the idea behind this approach is that we do not need to duplicate efforts in order to show users what is new with this release. 

The way that it is written it will only display this content if the release post includes a tag that matches the Jetpack version number exactly, so it's something that we can decide to add/remove on a per-release basis. Since we don't generally write release posts for minor changes (point releases), these will not be shown for minor changes.

MichaelArestad commented:
> the idea behind this approach is that we do not need to duplicate efforts in order to show users what is new with this release.

Do you mean with the changelog?

dereksmart commented:
@MichaelArestad I mean with the release post

MichaelArestad commented:
>  I mean with the release post

Ah. That might be more efficient for us, but it's not ideal for the user. I don't want them to see a modal every time they log in and are trying to do something. That's why it should be short, sweet, and only for major changes or major new features.

jeherve commented:
> I don't want them to see a modal every time they log in and are trying to do something. That's why it should be short, sweet, and only for major changes or major new features.

It's only displayed once after an update though; they should not see the modal again if they visit their dashboard once again. Do you think once per major version is still too much?

MichaelArestad commented:
> It's only displayed once after an update though; they should not see the modal again if they visit their dashboard once again. Do you think once per major version is still too much?

I think so, but I might be a little close to the project since I helped work on this feature. Calling in @joanrho and @jeffgolenski for a second opinion here.

joanrho commented:
Agree with @MichaelArestad that we should use transparent SVGs (no white bg behind illustration) and also keep things as short as possible by only highlighting major changes and linking to the release post for more details.

kbrown9 committed:
Upgrade notice: Remove hidden elements and include show_modal tag

 - Elements in the release posts with the class 'hide-in-jetpack' should
   not be displayed in the upgrade modal, so remove them from the content
   that will be displayed in the modal.
 - Add another tag, 'show_modal', that must be used on release posts that
   will be displayed in the upgrade modal.

enejb commented on class.jetpack.php:
```diff
@@ -3188,14 +3189,99 @@ public static function get_activation_source( $referer_url ) {
+		$content = wp_kses_post( $post_content );
+
+		// Remove the hidden elements from the modal content.
+		$dom_doc = new DOMDocument();
```
Can we check if this library is installed before we used it?
Since some host do not have it installed by default.

kbrown9 replied:
Good point - we should check before using it. I'll fix that.

kbrown9 replied:
@enejb  - I added a check for the DOMDocument class in aaf22fc. If the class isn’t installed, we just bail and don’t show the update notice at all.

Do you know if it’s common for this library to not be installed on hosts? If it is, we could remove the hidden elements on the front-end instead.

kbrown9 committed:
Update Notice: Use welcome svg image and show_udpate_modal tag

- We need to use a transparent SVG for the image in the modal. So, do not
  use the release post image. Instead, use the jetpack-welcome.svg image.
- Change the required release post tag to 'show_update_modal' for clarity.
- Update the docblock for 'get_release_post_data'.

gravityrail committed:
Merge branch 'master' into update/add_release_post_to_update_modal

kbrown9 committed:
Update Notice: Check for DOMDocument class

- Check for DOMDocument before using it.
- Change variable name for clarity.

kbrown9 commented:
@MichaelArestad @joanrho - thanks for your comments! I think my recent commits addressed all of them.

First, I switched the image to a transparent SVG. Since the release posts don’t use svgs, we’re now always using the jetpack-welcome.svg. I used a width of 250 for this image because that’s the width that was used previously. If you’d like a different size or image, let me know (see the screenshot below).

We’ve also put together a workflow to keep the update modal as short as possible while still being useful. Here’s the plan:

1) The release post author will need to give the release post two tags - the Jetpack version number and ‘show_update_modal’. If either of these tags are missing, the update modal will not display. So the author decides whether the update modal will display for each release.

2) The release post author can hide elements of the release post from the modal using the ‘hide-in-jetpack’ class. The author will be able to hide  things like the credits and any other elements they don’t want in the modal This will keep the modal focused on the most useful and important info.

Let us know if you have any suggestions about this!

Here's a screenshot of the image at the top of the modal: 

![Screen Shot on 2019-11-20 at 16-07-14](https://user-images.githubusercontent.com/50059399/69279147-c0346e80-0bb1-11ea-814d-3616d97a50b2.png)

gravityrail commented:
Got this to work after removing the query for the show_update_modal tag (didn't catch that in the thread, whoops). I am in favor of merging this.

IMO the release post looks good. Full size screenshot grabbed below for any designers who are having trouble testing this.

![goldsounds2 ngrok io_wp-admin_admin php_page=jetpack(iPad Pro) (3)](https://user-images.githubusercontent.com/51896/69283070-8c9d1880-0ba0-11ea-8da3-bed3cf597428.png)

gravityrail reviewed (dismissed):
Looks good to me.

keoshi commented:
I echo @MichaelArestad's comments in that this should ideally be short and sweet — otherwise we end up spamming customers every month. Just looking at the latest screen from @gravityrail that modal feels huge to me.

Would it be possible to add a “ghost” Gutenberg block in our blog that would:

- Not show up in the blog post.
- Contain all the necessary content + styling that gets pulled into this modal?

This would allow us to have the best of both worlds.

jeherve commented:
@keoshi How long do you think it should be? A couple of sentences, like an excerpt? More?

The screenshot above is a bit much indeed, but the modal would not include the contributor list, or really anything after "And More".

keoshi commented:
@jeherve That would depend on the release, I'm sure some would require a paragraph or two, while for others a bulleted list would suffice. While it might sound limited, highlighting just a few things brings more attention to them.

But my point was that it feels we need some control over what's displayed here; otherwise users won't benefit from this change at all since most will likely close that off immediately without reading any of it.

keoshi commented:
Adding here a quick wireframe of how I'm picturing this to become more useful:

![image](https://user-images.githubusercontent.com/390760/70542495-b4105100-1b60-11ea-9db7-77e43afca3eb.png)

jeherve commented:
I see what you mean. We don't quite follow this structure in the release post though, as we don't always have the material to dot that. I don't know that we can expect a similar structure in those posts every month.

keoshi commented:
That’s why I suggested adding a ghost block. It could live side by side with the post’s content and it would only show up in here.

folletto commented:
Ok, here's a proposal to move this forward. Will require some code adjustment, but seems a viable way.

Design, let's use the most recent proposals put forward for search, and make them a bit more generic so they can work with any content:
![announcement-modal-i2](https://user-images.githubusercontent.com/4389/77758526-0485be80-702b-11ea-863d-422b5b6146fd.png)

Note: the image loaded is the featured image. If missing that entire block disappears, and the X remains on white, with text.

Functionally, I did some brainstorming with @jeherve and @robertbpugh, here's the idea:

1. We don't use content from Jetpack Blog.
2. We create a new private site that manages only and exclusively these modals.
3. Every time a post is pushed live on that site, it gets shown in WP Admin as per design above.
4. The limit is once per day, we don't want to check too often (this is up to discussion, let's do what's sensible and quick to ship).
5. The new site is owned by Marketing (Rob), which will craft short and to-the-point messages when a release is deemed worth of it.

gravityrail commented:
@folletto since that is a much more substantial and technically different change from this one, I am going to close this PR.

gravityrail committed:
Merge branch 'master' into update/add_release_post_to_update_modal

gravityrail commented:
@folletto I implemented the design in my branch. Here's some screenshots, with and without a featured image. I use Photon to scale the featured image - this PR assumes it's hosted on WPCOM servers so the resize params work.

Pixel 2 XL, no image:

![upgrade-dialog-pixel-2xl-noimage](https://user-images.githubusercontent.com/51896/78195488-9713cc80-7434-11ea-84c9-1a12deb28412.png)

Pixel 2 XL, with image:

![upgrade-dialog-pixel-2xl](https://user-images.githubusercontent.com/51896/78195505-9da24400-7434-11ea-8e09-55f80a4c82f1.png)

iPad, no image:

![upgrade-dialog-ipad-noimage](https://user-images.githubusercontent.com/51896/78195519-a561e880-7434-11ea-90c2-775ac63242de.png)

iPad, with image:

![upgrade-dialog-ipad](https://user-images.githubusercontent.com/51896/78195530-abf06000-7434-11ea-9b2c-022552c526d3.png)

iPad Pro, horizontal, no image:

![upgrade-dialog-ipadpro-horizontal-noimage](https://user-images.githubusercontent.com/51896/78195552-c0345d00-7434-11ea-926a-18b3eda1a6de.png)

iPad Pro, horizontal, with image:

![upgrade-dialog-ipadpro-horizontal](https://user-images.githubusercontent.com/51896/78195558-c9bdc500-7434-11ea-8bd2-f677b5b2d7c8.png)

gravityrail committed:
Create new component, styles and behaviours to match design

gravityrail committed:
Restore default version

folletto commented:
I think it's solid, just needs some mostly visual tweaks.

Just a couple of tweaks:
1. The wide shadow needs to be added.
2. The width should be fixed.
3. The buttons and links at the bottom should be removed.


Removing these buttons, to be clear: 
![Screenshot 2020-04-02 at 16 49 23](https://user-images.githubusercontent.com/4389/78269906-16230680-7502-11ea-9593-8c3773bd2f94.png)

kbrown9 committed:
JetpackStateNotice / UpgradeNoticeContent: update components

Update the JetpackStateNotice component:
 - Check that the 'messageContent' variable is not null before using its
   'release_post_content' property. The value of 'messageContent' will
   normally be null.
 - Remove the adminURL prop because it's not used.
 - Remove the isUnavailableInDevMode prop because it's not used.
 - Clean up the dependencies.

Update the UpgradeNoticeContent component:
 - Remote the dismiss button and changelog link.
 - Clean up the dependencies.

kbrown9 committed:
Update Modal: change width and add shadow

kbrown9 committed:
jp-dialogue-modern-full_container: increase z-index

Increase the z-index so the modal sits overs the jetpack-connect container.

kbrown9 commented:
The following changes were made in the latest commits:

1. Added a shadow.
2. Adjusted the width to a maximum of 300 px.
3. Removed the button and link from the bottom of the modal.

**Designers:** If you’d like access to a test site that displays the modal, contact me.

Screenshot of the latest changes on desktop:

![Screen Shot on 2020-04-07 at 15-01-06](https://user-images.githubusercontent.com/50059399/78709380-87d3c980-78e1-11ea-874e-fbb2c44f1794.png)

keoshi commented:
Uhh, that is looking mighty good!

folletto commented:
I did a test run, trying different lengths, trying it on smaller and larger screens. I think there's just one more tweak and then we can merge:

The title seems always "New in Jetpack 8.3", and doesn't seem to pick up the title from the post. Can we do that? :)

I also noticed there's some minor thing in the styling:
1. the text is gray
2. the bullet list is smaller

It's probably just a matter of tweaking `.jp-dialogue-modern p` and `.jp-dialogue-modern li`, having the text `#000000` and a size `16px`. However, are these used elsewhere too? — I want to avoid to break anything else :D 

I'd also restore the underline (`text-decoration`) to `a` tags.

Example with tweaked fonts: 
![Screenshot 2020-04-08 at 12 29 39](https://user-images.githubusercontent.com/4389/78779309-a777fa00-7994-11ea-8f54-b5a0cdf8e142.png)

kbrown9 committed:
Update Modal: Use post title and improve design

- Use the release post title in the modal. If the release post doesn't include
  a title, default to "New in Jetpack {version}".

- Improve the design:
  - jp-dialogue-modern p and li: font-size and color
  - jp-dialogue-modern a: text-decoration

- Clean up Jetpack::send_update_modal_data():
  - Since we're now using a post that has been written for the modal, there's no
    need to remove hidden elements. Remove the DOM traversal that looked for
    hidden elements.

kbrown9 committed:
Update Modal: revert jetpack-dialogue style changes

Revert the jetpack-dialogue style changes. The modal is now using
the ModernOverlay component rather than the JetpackDialogue component.

kbrown9 commented:
Here are a couple of screenshots to show the latest updates to the design. 

iPad Pro, horizontal, with a featured image:
![Screen Shot on 2020-04-10 at 02-35-01](https://user-images.githubusercontent.com/50059399/78969297-96ce9d80-7ad4-11ea-87e0-039e59a2d632.png)

iPad Pro, horizontal, without a featured image:
![Screen Shot on 2020-04-10 at 02-36-50](https://user-images.githubusercontent.com/50059399/78969274-8d453580-7ad4-11ea-9ac4-faa693c4f79b.png)

Pixel 2 XL without a featured image (I haven't had a chance to adjust the top margin on small devices yet.)
<img src=https://user-images.githubusercontent.com/50059399/78969460-faf16180-7ad4-11ea-8d1d-b3634cc26caf.png width=250>

folletto commented:
Wonderful! Design wise, all good now.  👍 
 
Let's get this shipped! :D

kbrown9 commented:
This is ready for review!

In a [comment above](https://github.com/Automattic/jetpack/pull/13710#issuecomment-604989500), this process for deciding when to display the modal was proposed:

> 	3.	Every time a post is pushed live on that site, it gets shown in WP Admin as per design above.
> 	4.	The limit is once per day, we don't want to check too often (this is up to discussion, let's do what's sensible and quick to ship).

This isn’t how the modal currently works. Right now, we check for a release post only when the plugin is updated to a new version. So, in between version updates, we don’t check for release posts.

I’d like to handle the new proposed process in a separate PR for two reasons:
- To ensure that this version of the release announcement modal is shipped in 8.5.
- To allow reviewers of that PR to focus on the new modal display decision process.

Also, I didn’t work on adjusting the top margin on small devices yet. Again, I think we can handle that in a separate PR.

folletto commented:
> I’d like to handle the new proposed process in a separate PR for two reasons: [...]

Yes. I agree. I always endorse smaller incremental works, and this is better if it gets in first. :)

> Also, I didn’t work on adjusting the top margin on small devices yet.

Not major, but again, yes. :)

kraftbj commented:
One flow issue I found that I think is unintentional.

With this branch, I followed the testing instructions:
* Set version to 8.2.
* Load JP Dash.
* Set version to 8.3.
* Reload JP Dash.

This worked as expected and I saw the modal.

If I do the same except the last step:
* Set version to 8.2.
* Load JP Dash.
* Set version to 8.3.
* Do other things in wp-admin.
* Visit JP Dash.

It doesn't appear. Skimming the code, I believe since we're using the state, it will only keep for the next request. I know we don't want it to show too much, but wanting to confirm that is indeed the expected (only if they visit the dashboard immediately after upgrading).

folletto commented:
No, it should always show once per release (if we decide to show it) — so you're correct, it's not an intentional choice.

That said, let's merge this PR first because it works and it doesn't show up broken anywhere — then let's review and fix that behaviour.

kbrown9 commented:
> If I do the same except the last step:
> 
> * Set version to 8.2.
> * Load JP Dash.
> * Set version to 8.3.
> * Do other things in wp-admin.
> * Visit JP Dash.
> 
> It doesn't appear. Skimming the code, I believe since we're using the state, it will only keep for the next request. I know we don't want it to show too much, but wanting to confirm that is indeed the expected (only if they visit the dashboard immediately after upgrading).

Thanks for thoroughly testing this! You're right about the issue being the state. We're not [putting the release post content](https://github.com/Automattic/jetpack/pull/13710/files#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R5873) in the `jetpackState` cookie, so the`message_content` state isn't persisted to the JP dashboard load. 

What do you think of merging the PR as is and fixing that in a separate PR?

dereksmart commented:
> One flow issue I found that I think is unintentional.

@kraftbj I believe Kim had already addressed that in [this comment](https://github.com/Automattic/jetpack/pull/13710#issuecomment-612116922), and Erin agreed it's fine for this round. I don't see any reason why this PR can't ship in 8.5?

dereksmart reviewed (approved):
See comment above ^

kraftbj commented:
No objection to landing in 8.5 since we're acknowledged if the flow I tried was intentional or not and, since not, we intend to follow up. 

If it isn't something we'd fix in beta (thus not updated until 8.6), I'd like an issue opened just so it doesn't fall off the map but that's IMO.

jeherve commented:
Cherry-picked to `branch-8.5` in 4510f35b8084be0046a74e4fc4501a4b21886bc4
